### PR TITLE
[ODS-6126] Use GitHub Action from Docker Scout to analyze docker images

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Docker/Invoke-Build.ps1 [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (simpat-jesus) [inline Docker/Invoke-Build.ps1:1]: This duplicates https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/blob/main/Docker/build-images.ps1

I think we need to refactor both scripts, but that can be done in another ticket

File: .github/workflows/Analyze docker images.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          image: ${{ env.IMAGENAME }}
+          sarif-file: sarif-${{ matrix.dockerfile.name }}.${{ env.PACKAGE_VERSION }}.0-${{ matrix.StandardVersion }}.output.json
+          summary: true
+          only-severities: "critical,high"
+      - name: Upload SARIF result
+        id: upload-sarif
+        if: ${{ github.event_name != 'pull_request_target' }}
+        uses: github/codeql-action/upload-sarif@df32e399139a3050671466d7d9b3cbacc1cfd034 #codeql-bundle-v2.15.2
+        with:
+          sarif_file: sarif-${{ matrix.dockerfile.name }}.${{ env.PACKAGE_VERSION }}.0-${{ matrix.StandardVersion }}.output.json
```
Comment (simpat-jesus) [inline .github/workflows/Analyze docker images.yml:122]: The current edfialliance/ods-api-web-api image has a vulnerability, but the summary only displays the info for 5.0.0, not sure if this is a buf or expected from the github/codeql-action/upload-sarif action

**Review by simpat-jesus — CHANGES_REQUESTED** [review]

File: .github/workflows/Analyze docker images.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  cancel-in-progress: true
+
+env:
+  CURRENT_STANDARD_VERSION: "5.0.0"
+  PACKAGE_VERSION: "7.2"
+  DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+  DOCKER_USERNAME: ${{ vars.DOCKER_USERNAME }}  
+
+jobs:
+  FindStandardAndExtensionVersions:
```
[the remark this answers is not in the document]
Comment (semalaiappan) [reply .github/workflows/Analyze docker images.yml:24]: Reuseable Action updated. Thanks

File: Docker/Invoke-Build.ps1 (hunk 1 above)
Comment (semalaiappan) [reply Docker/Invoke-Build.ps1]: removed Build images ps1 file and  now all places invoked by invoke build ps1 file

File: .github/workflows/Analyze docker images.yml (hunk 2 above)
Comment (semalaiappan) [reply .github/workflows/Analyze docker images.yml:83]: To view all Vulnerabilities  here . you need to click edited icon . display only show most recent one

File: .github/workflows/Docker Test.yml [hunk 4]
```diff
@@ -63,17 +63,28 @@ jobs:
...
+            { name: "ods-api-db-admin", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-minimal", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-sandbox", path: "alpine/pgsql" },
+            { name: "ods-api-swaggerui", path: "alpine" },
+            { name: "ods-api-web-api", path: "alpine/pgsql" },
+            { name: "ods-api-web-api", path: "alpine/mssql" },            
+            { name: "ods-api-web-sandbox-admin", path: "alpine/pgsql" },
+            { name: "ods-api-web-sandbox-admin", path: "alpine/mssql" }            
+          ]        
     name: Docker Build (Standard ${{ matrix.StandardVersion }} Extension ${{ matrix.ExtensionVersion }})
```
Comment (simpat-jesus) [inline .github/workflows/Docker Test.yml:77]: We should add the image name in the job name, something like ImageName (Std, Ext) to make it easy finding which image we are looking at and not a list of 'Docker Build ...'
image

File: .github/workflows/Publish to Docker Hub.yml [hunk 5]
```diff
@@ -47,6 +47,17 @@ jobs:
...
+            { name: "ods-api-db-admin", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-minimal", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-sandbox", path: "alpine/pgsql" },
+            { name: "ods-api-swaggerui", path: "alpine" },
+            { name: "ods-api-web-api", path: "alpine/pgsql" },
+            { name: "ods-api-web-api", path: "alpine/mssql" },            
+            { name: "ods-api-web-sandbox-admin", path: "alpine/pgsql" },
+            { name: "ods-api-web-sandbox-admin", path: "alpine/mssql" }            
+          ]        
     name: Publish to Docker Hub (Standard ${{ matrix.StandardVersion }} Extension ${{ matrix.ExtensionVersion }})
```
Comment (simpat-jesus) [inline .github/workflows/Publish to Docker Hub.yml:61]: Include the image name to the job name

File: .github/workflows/Analyze docker images.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          [
+            { name: "ods-api-db-admin", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-minimal", path: "alpine/pgsql" },
+            { name: "ods-api-db-ods-sandbox", path: "alpine/pgsql" },
+            { name: "ods-api-swaggerui", path: "alpine" },
+            { name: "ods-api-web-api", path: "alpine/pgsql" },
+            { name: "ods-api-web-api", path: "alpine/mssql" },            
+            { name: "ods-api-web-sandbox-admin", path: "alpine/pgsql" },
+            { name: "ods-api-web-sandbox-admin", path: "alpine/mssql" }            
+          ]
```
Comment (simpat-jesus) [inline .github/workflows/Analyze docker images.yml:46]: Add a job name that includes the image name

**Review by simpat-jesus — CHANGES_REQUESTED** [review]

File: .github/workflows/Docker Test.yml (hunk 4 above)
Comment (semalaiappan) [reply .github/workflows/Docker Test.yml]: Updated . Thanks

File: .github/workflows/Publish to Docker Hub.yml (hunk 5 above)
Comment (semalaiappan) [reply .github/workflows/Publish to Docker Hub.yml]: Updated . Thanks

File: .github/workflows/Analyze docker images.yml (hunk 6 above)
Comment (semalaiappan) [reply .github/workflows/Analyze docker images.yml:46]: Updated . Thanks

**Review by simpat-jesus — APPROVED** [review]

**Review by vimayya — APPROVED** [review]

**Pull request merged**